javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -219,7 +219,9 @@ public Map<Request, List<ProxyMethod>> map(Operation operation) {
                     final boolean parameterIsRequired = parameter.isRequired();
                     final boolean parameterIsClientOrApiVersion = ClientModelUtil.getClientDefaultValueOrConstantValue(parameter) != null
                             && parameter.getLanguage().getJava().getName().equalsIgnoreCase("apiversion");
-                    if (parameterIsRequired || parameterIsClientOrApiVersion) {
+                    if (parameterIsRequired
+                            || !parameterIsRequired && (proxyMethodParameter.getIsConstant() || proxyMethodParameter.getFromClient())
```

Comment:
Use similar patten as e.g. `parameterIsRequired`. use a local variable with clear name for this condition of parameter (and remove `!parameterIsRequired`, this is not a condition for your case).